sui-execution/v0/sui-adapter/src/temporary_store.rs

```diff
@@ -532,7 +532,16 @@ impl<'backing> TemporaryStore<'backing> {
                     // us from catching this.
                 }
                 Owner::ObjectOwner(_parent) => {
-                    unreachable!("Input objects must be address owned, shared, or immutable")
+                    unreachable!(
+                        "Input objects must be address owned, shared, consensus, or immutable"
+                    )
+                }
+                Owner::ConsensusV2 { authenticator, .. } => {
+                    assert!(
+                        sender == authenticator.as_single_owner(),
```

Comment:
i think we can leave this `unimplemented!()` for old execution versions - they will never see these objects.

Reply:
done